src/Command/ModuleDeployAfterMagentoCommand.php

```diff
@@ -43,83 +112,152 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $startTime = microtime(true);
 
-        $projectRoot = getcwd();
-        $directoryName = basename($projectRoot);
+        $mainDomain = basename(getcwd());
+        $modulesFolder = $input->getArgument(self::OPTION_FOLDER);
+
+        $together = $input->getOption(self::OPTION_TOGETHER);
 
-        # Stage 1: copy files, deploy Sample Data, create database
-        if (!file_exists('./app') || !is_dir('./app')) {
-            $output->writeln('<error>Seems we\'re not inside the Magento project</error>');
-            exit;
+        if (file_exists('./composer.json')) {
+            $magentoComposerFile = json_decode(file_get_contents('./composer.json'));
+
+            if (!is_dir('./app') || strpos($magentoComposerFile->name, 'magento/') === false) {
+                $output->writeln('<error>Seems we\'re not inside the Magento project</error>');
+                exit(1);
+            }
+        } else {
+            $output->writeln('<error>Composer.json not found</error>');
```

Comment:
@serhiira 
1) Exception must explain why this is a problem, not just be thrown with some text. Must be something like `Seems we're not in the root folder of the installed Magento 2 application.`
2) Move this validation to some method.